clients/java/src/test/java/io/camunda/zeebe/client/impl/worker/JobWorkerMetricsTest.java

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.zeebe.client.impl.worker;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.camunda.zeebe.client.api.JsonMapper;
+import io.camunda.zeebe.client.api.response.ActivatedJob;
+import io.camunda.zeebe.client.api.worker.JobWorkerMetrics;
+import io.camunda.zeebe.client.impl.ZeebeObjectMapper;
+import io.camunda.zeebe.client.impl.response.ActivatedJobImpl;
+import java.time.Duration;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.BooleanSupplier;
+import java.util.function.Consumer;
+import java.util.function.IntConsumer;
+import org.jmock.lib.concurrent.DeterministicScheduler;
+import org.junit.jupiter.api.Test;
+
+final class JobWorkerMetricsTest {
+
+  private final DeterministicScheduler executor = new DeterministicScheduler();
+
+  @Test
+  void shouldCountActivatedStreamedJobs() {
+    // given
+    final TestJobStreamer streamer = new TestJobStreamer();
+    final TestJobWorkerMetrics metrics = new TestJobWorkerMetrics();
+
+    try (final JobWorkerImpl ignored = createWorker(0, streamer, metrics)) {
+      // when
+      streamer.streamJob();
+      streamer.streamJob();
+
+      // then
+      executor.runUntilIdle();
+      assertThat(metrics.jobsActivated).hasValue(2);
+    }
+  }
+
+  @Test
+  void shouldCountHandledStreamedJobs() {
+    // given
+    final TestJobStreamer streamer = new TestJobStreamer();
+    final TestJobWorkerMetrics metrics = new TestJobWorkerMetrics();
+
+    try (final JobWorkerImpl ignored = createWorker(2, streamer, metrics)) {
+      // when
+      streamer.streamJob();
+      streamer.streamJob();
+      streamer.streamJob();
+
+      // then
+      executor.runUntilIdle();
+      assertThat(metrics.jobsHandled).hasValue(2);
+    }
+  }
+
+  @Test
+  void shouldCountActivatedPolledJobs() {
+    // given
+    final TestJobPoller poller = new TestJobPoller();
+    final TestJobWorkerMetrics metrics = new TestJobWorkerMetrics();
+
+    try (final JobWorkerImpl ignored = createWorker(0, poller, metrics)) {
+      // when
+      executor.tick(1, TimeUnit.MINUTES);
+      poller.streamJob();
+      poller.streamJob();
```

Comment:
The stream job name comes from the fact both are gRPC streams (whether polling or not), but I can see how that's confusing :smile: What about `produceJob`? No verb is particularly good for a poller, since here we want to send a job to the poll consumer.

Reply:
> ❌ it feels to me that this test does not verify what is described by its name. Could you please check for me that this is the expected implementation? I worry that this tests the job streaming instead.

We're explicitly testing polling here because we actually invoke the JobPoller's consumer directly when producing the job. So we don't touch the `TestJobStreamer` consumer at all. Does that make sense? 

One of your comments earlier was about fixing a bug within the `JobWorker` (see https://github.com/camunda/zeebe/pull/13818#discussion_r1296072083). This means there are effectively two different consumers, even if they end up using the same factory. If you're not sure, you can debug the test and put a breakpoint on `JobWorkerImpl#handleStreamedJob` and `JobWorkerImpl#handleJob`, the two effective consumers, and see which is called.